### ISSUE TITLE
Have to move a sender to thread otherwise recv() fails

### ISSUE THREAD

**Issue by Sibz** [crossbeam-rs/crossbeam#570]
```rust
        let receiver = self.thread_signal_channels.1.clone();
        let sender = self.thread_signal_channels.0.clone();
        std::thread::spawn(move || {
            let _s = sender; // TODO this shouldn't be needed
            loop {
                match receiver.recv()
                {
```
The thread has no need for the sender, however the recv fails if I don't move it to the closure. The original sender is safely contained in a struct.
I was hoping that once all senders were closed the recv would err and I could end the thread execution.
Unfortunately moving it to the thread means it will never err and the thread lives on eternally.

Perhaps I am missing an important concept, but I feel like the recv should not fail in when the sender isn't moved to the thread because the original sender exists.

**Comment by Sibz** [comment]
Can anyone shed light on this?

**Comment by taiki-e** [comment]
Could you provide code to reproduce this issue?

**Comment by Sibz** [comment]
After thinking about it for a while, only the senders clone is used later on in the code, and the original sender is unused. Even though the original sender is assigned to a field, it must not 'register' as it's unused in the place of declaration and a clone is sent off for use in a thread later.
My guess is I have use the original sender somewhere, not just clones.

I'll try create a mock up that reproduces this.

**Comment by Sibz** [comment]
Ok, I was unable to reproduce. I went back to the code where I had to assigned sender inside the thread and removed it, and it all works fine now.

My guess is during development I hadn't yet implemented the sender side of the channel completely and the sender was in fact being dropped. Though I specifically remember going through checking that the sender was not dropped. However it's a complex bit of code so I may have overlooked something.

As I cannot reproduce, and managed to remove this uneccessary assignment without side effects in the original problematic code, I will close this issue.
